### DISCUSSION TITLE
PDF export

### DISCUSSION THREAD

**Discussion by valeriolg** [danielsaidi/RichTextKit#182] [Q&A]
Hello, how can I achieve to export the text into PDF? I see that there are some methods and export buttons, but I don't understand how to use it. Thanks, Valerio.

**Comment by danielsaidi** [comment]
There are various tools you can use to achieve this, but perhaps they need to be better explained in the documentation.

**Reply by valeriolg** [reply]
Hi @danielsaidi I eventually found how to export to PDF digging deep into the documentation. I would suggest a bit more focus on this topic in the documentation as I believe it's an important feature.
On another topic, in macOS 14 I have problems with the textview cursor color not being visible on a white background. My app faces dark mode and I've successfully forced the RichTextEditorView to light mode to have it visible on macOS 13, but on macOS 14 seems it's not working. Any advice? Thanks, Valerio.

**Comment by danielsaidi** [comment]
Happy you got it working @valeriolg! I will keep this discussion around and add more information about the various exporting options the next time I work on the library.

I haven't noticed this in macOS 14. It shouldn't be like that, since the editor just wraps a native text view. Does a regular text view behave in the same way?

**Comment by karlmira** [comment]
Here's my code I use for creating a PDF unter Swift and SwiftUI. I hope, it will help you and all the others, that need this as a base for their own code.
[Letter2Pdf.swift.zip](https://github.com/user-attachments/files/17045938/Letter2Pdf.swift.zip)
